### DISCUSSION TITLE
Showcase

### DISCUSSION THREAD

**Discussion by abetlen** [abetlen/llama-cpp-python#25] [Show and tell]
Share anything cool you've built with `llama-cpp-python` here

**Comment by jediknight813** [comment]
I made a discord chatbot using this that people can easily use and customize: https://github.com/jediknight813/llama-cpp-discord-bot

**Comment by zeyus** [comment]
I've made a RAG assistant using `llama-cpp-python` and `langchain` that supports dropping in multiple models and allows you to load a set of markdown notes (e.g. from Obsidian or similar) and ask the model about what you have written. The core functionality is all working and I'm currently evaluating between different lightweight models to get some results for one of my exam papers :)

https://github.com/zeyus/RAGGA
